Automattic/wp-calypso
# Seller Experience: Use signup infrastructure rather than a hard-coded redirect

#### Changes proposed in this Pull Request

* Switch from using `page.redirect` at the end of the seller experience simple flow to using `submitSignupStep()` and `goToNextStep()`.
* This PR adds an API callback to the `store-features` step that fetches information about the site (checking to see if FSE is active) and uses the data to determine where to send the user via the `getDestinationFromIntent` callback that fires when the flow is complete.
* This more closely follows what other signup flows do and will make it slightly easier to update the flow to add a design picker step, for example. 

#### Testing instructions

* Switch to this PR, navigate to `/start`, and make sure `?flags=seller-experience` is enabled on your local dev environment
* Go through the signup flow; create a new free site.
* At the intent step, select "Sell"
* Add your site info
* Choose "Start simple"
* You should be redirected to the site editor.
* Go through the same flow with an existing site that uses an older theme by visiting `/start/setup-site/intent?siteSlug=yourtestsite.wordpress.com&flags=seller-experience` replacing yourtestsite with your site slug
* Finish the seller experience flow via the Simple route and you should be redirected to the page editor instead of the site editor.

Related to #60772
